CMakeLists.txt

```diff
@@ -44,6 +47,24 @@ set(SEALAPI_SOURCES
 pybind11_add_module(_sealapi_cpp ${SEALAPI_SOURCES})
 
 if(NOT WIN32)
+    target_link_libraries(libtenseal PRIVATE seal)
     target_link_libraries(_tenseal_cpp PRIVATE seal)
     target_link_libraries(_sealapi_cpp PRIVATE seal)
 endif()
+
+################################
+# Unit Tests
+################################
+ADD_SUBDIRECTORY (third_party/googletest)
+enable_testing()
+include_directories(${gtest_SOURCE_DIR}/include ${gtest_SOURCE_DIR} ${gmock_SOURCE_DIR}/include third_party/SEAL)
+
+set(TENSEAL_TESTS_BASEDIR tests/cpp)
+set(TESTING_SOURCES
+    ${TENSEAL_TESTS_BASEDIR}/tensealcontext_test.cpp
+    ${TENSEAL_TESTS_BASEDIR}/tensors/ckksvector_test.cpp
+    ${TENSEAL_TESTS_BASEDIR}/tensors/bfvvector_test.cpp
+)
+add_executable(tenseal_tests ${TESTING_SOURCES} ${SOURCES} )
+target_link_libraries(tenseal_tests PRIVATE gtest gtest_main seal libtenseal)
+add_test( tenseal_tests tenseal_tests )
```

Comment:
This whole block also could be conditioned by an environment variable BUILD_TEST

Reply:
fixed